From 86a50d6bad2ba44a8107ddb379d2ec6b84537a48 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Tue, 16 May 2017 11:58:12 +0200
Subject: [PATCH 3/3] fix build error with gcc -std=c++98
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

atom-object.cxx: In member function ‘bool {anonymous}::MatchLink::operator()(AtomLink)’:
atom-object.cxx:63:91: error: no matching function for call to ‘remove_if(std::__cxx11::basic_string<char>::iterator, std::__cxx11::basic_string<char>::iterator, <unresolved overloaded function type>)’
          linkType.erase( remove_if( linkType.begin(), linkType.end(), std::isspace ), linkType.end() );
---
 src/libcmis/atom-object.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libcmis/atom-object.cxx b/src/libcmis/atom-object.cxx
index 01454a4..510f013 100644
--- a/src/libcmis/atom-object.cxx
+++ b/src/libcmis/atom-object.cxx
@@ -30,6 +30,7 @@
 #include <locale>
 #include <sstream>
 
+#include <boost/algorithm/string.hpp>
 #include <boost/date_time/posix_time/posix_time.hpp>
 
 #include "atom-document.hxx"
@@ -59,7 +60,7 @@ namespace
                 // Some implementations (xcmis) put extra spaces into the type attribute
                 // (e.g. "application/atom+xml; type=feed" instead of "application/atom+xml;type=feed")
                 string linkType = link.getType( );
-                linkType.erase( remove_if( linkType.begin(), linkType.end(), isspace ), linkType.end() );
+                linkType.erase( remove_if( linkType.begin(), linkType.end(), boost::is_space() ), linkType.end() );
 
                 // Some implementation (SharePoint) are omitting the type attribute
                 bool matchesType = m_type.empty( ) || linkType.empty() || ( linkType == m_type );
-- 
2.9.4

